kevm-pyk/src/kevm_pyk/kproj/evm-semantics/lemmas/evm-int-simplification.k

```diff
@@ -65,6 +65,11 @@ module EVM-INT-SIMPLIFICATION-COMMON
     requires 2 ^Int ( lengthBytes(BA) *Int 8) <=Int X
     [concrete(X), simplification]
 
+  rule [asWord-lt-concat]:
+    #asWord ( BA1 +Bytes BA2 ) <Int X => #asWord ( BA1 ) <Int ( X -Int #asWord ( BA2 ) ) /Int ( 2 ^Int ( 8 *Int lengthBytes ( BA2 ) ) )
+    requires ( X -Int #asWord ( BA2 ) ) modInt ( 2 ^Int ( 8 *Int lengthBytes ( BA2 ) ) ) ==Int 0
+    [concrete(BA2, X), simplification]
```

Comment:
This is an auxiliary simplification required for the interaction of `[shift-as-asWord]` and `[buf-asWord-invert-lr-len-gt]` so that the expressivity of `[buf-shift]` is preserved.